otx/algorithms/classification/adapters/deep_object_reid/utils/utils.py

```diff
@@ -0,0 +1,396 @@
+from otx.api.utils.argument_checks import (
+    DatasetParamTypeCheck,
+    check_input_parameters_type,
+)
+
+
+@check_input_parameters_type()
+def generate_label_schema(not_empty_labels: List[LabelEntity], multilabel: bool = False):
+    """Generate label schema."""
+    assert len(not_empty_labels) > 1
+
+    label_schema = LabelSchemaEntity()
+    if multilabel:
+        emptylabel = LabelEntity(name="Empty label", is_empty=True, domain=Domain.CLASSIFICATION)
+        empty_group = LabelGroup(name="empty", labels=[emptylabel], group_type=LabelGroupType.EMPTY_LABEL)
+        for label in not_empty_labels:
+            label_schema.add_group(LabelGroup(name=label.name, labels=[label], group_type=LabelGroupType.EXCLUSIVE))
+        label_schema.add_group(empty_group)
+    else:
+        main_group = LabelGroup(name="labels", labels=not_empty_labels, group_type=LabelGroupType.EXCLUSIVE)
+        label_schema.add_group(main_group)
+    return label_schema
+
+
+# pylint: disable=invalid-name, too-many-locals
+@check_input_parameters_type()
+def get_multihead_class_info(label_schema: LabelSchemaEntity):
+    """Get multihead class info by label schema."""
+    all_groups = label_schema.get_groups(include_empty=False)
+    all_groups_str = []
+    for g in all_groups:
+        group_labels_str = [lbl.name for lbl in g.labels]
+        all_groups_str.append(group_labels_str)
+
+    single_label_groups = [g for g in all_groups_str if len(g) == 1]
+    exclusive_groups = [sorted(g) for g in all_groups_str if len(g) > 1]
+    single_label_groups.sort(key=itemgetter(0))
+    exclusive_groups.sort(key=itemgetter(0))
+    class_to_idx = {}
+    head_idx_to_logits_range = {}
+    num_single_label_classes = 0
+    last_logits_pos = 0
+    for i, group in enumerate(exclusive_groups):
+        head_idx_to_logits_range[i] = (last_logits_pos, last_logits_pos + len(group))
+        last_logits_pos += len(group)
+        for j, cls in enumerate(group):
+            class_to_idx[cls] = (i, j)  # group idx and idx inside group
+            num_single_label_classes += 1
+
+    # other labels are in multilabel group
+    for j, group in enumerate(single_label_groups):
+        class_to_idx[group[0]] = (len(exclusive_groups), j)
+
+    all_labels = label_schema.get_labels(include_empty=False)
+    label_to_idx = {lbl.name: i for i, lbl in enumerate(all_labels)}
+
+    mixed_cls_heads_info = {
+        "num_multiclass_heads": len(exclusive_groups),
+        "num_multilabel_classes": len(single_label_groups),
+        "head_idx_to_logits_range": head_idx_to_logits_range,
+        "num_single_label_classes": num_single_label_classes,
+        "class_to_group_idx": class_to_idx,
+        "all_groups": exclusive_groups + single_label_groups,
+        "label_to_idx": label_to_idx,
+    }
+    return mixed_cls_heads_info
+
+
+# pylint: disable=too-many-instance-attributes
+class ClassificationDataset:
```

Comment:
It would be desirable to have this in a separate data folder like other adapters or tasks. Not suitable for utils/utils.py.